File: src/main/scala/org/xbery/artbeams/common/assets/domain/Validity.scala
package org.xbery.artbeams.common.assets.domain

import java.time.Instant

import org.xbery.artbeams.common.assets.domain.AssetAttributes.EmptyDate

/**
  * Valid from and valid to attributes.
  * @author Radek Beran
  */
@SerialVersionUID(-1L)
case class Validity(
  validFrom: Instant,
  validTo: Option[Instant]
) extends Serializable {

  def updatedWith(validity: EditedTimeValidity): Validity = {
    this.copy(validFrom = validity.validFrom.toInstant(), validTo = validity.validTo.map(d => d.toInstant()))
  }
}

object Validity {

  lazy val Empty = Validity(
    EmptyDate,
    None
  )
}

File: src/main/kotlin/org/xbery/artbeams/common/assets/domain/Validity.kt
package org.xbery.artbeams.common.assets.domain

import org.xbery.artbeams.common.assets.domain.AssetAttributes.Companion.EmptyDate
import java.io.Serializable
import java.time.Instant

/**
 * Valid from and valid to attributes.
 * @author Radek Beran
 */
data class Validity(val validFrom: Instant, val validTo: Instant?) : Serializable {
    fun updatedWith(validity: EditedTimeValidity): Validity {
        val validTo = validity.validTo
        return this.copy(validFrom = validity.validFrom.toInstant(), validTo = validTo?.toInstant())
    }

    companion object {
        val Empty: Validity = Validity(EmptyDate, null)
    }
}
